src/engine/blocks.js

```diff
@@ -316,6 +317,14 @@ class Blocks {
                 if (variable) {
                     block.fields[args.name].value = variable.name;
                     block.fields[args.name].id = args.value;
+                } else {
```

Comment:
This whole variable logic is super weird, and I feel like it would be easy to miss bugs. Is it possible to change the source of this information to provide information in a cleaner way, where it always provides an ID and the value field isn't different things?

Reply:
Oh jeez sorry this is totally unrelated, didn't mean for it to be part of this PR.